libqtile/backend/wayland/xdgwindow.py

```diff
@@ -176,16 +106,63 @@ def _on_set_app_id(self, _listener: Listener, _data: Any) -> None:
         if self.ftm_handle:
             self.ftm_handle.set_app_id(self._wm_class or "")
 
-    def hide(self) -> None:
-        if self.mapped:
-            self.surface.unmap_event.emit()
-
     def unhide(self) -> None:
-        if not self.mapped:
-            self.surface.map_event.emit()
+        self._wm_class = self.surface.toplevel.app_id
+
+        if self not in self.core.pending_windows:
+            # Regular usage
+            if not self.container.node.enabled and self.group and self.group.screen:
+                self.container.node.set_enabled(enabled=True)
+                self.core.focus_window(self)
+            return
+
+        # This is the first time this window has mapped, so we need to do some initial
+        # setup.
+        self.core.pending_windows.remove(self)
+        self._wid = self.core.new_wid()
+        logger.debug(
+            "Managing new top-level window with window ID: %s, app_id: %s",
+            self._wid,
+            self._wm_class,
+        )
+
+        # Save the client's desired geometry
+        surface = self.surface
+        geometry = surface.get_geometry()
+        self._width = self._float_width = geometry.width
+        self._height = self._float_height = geometry.height
+
+        # Tell the client to render tiled edges
+        surface.set_tiled(EDGES_TILED)
+
+        handle = self.ftm_handle
+        assert handle is not None
+
+        # Get the client's name
+        if surface.toplevel.title:
+            self.name = surface.toplevel.title
+            handle.set_title(self.name)
+        if self._wm_class:
+            handle.set_app_id(self._wm_class or "")
+
+        # Add the toplevel's listeners
+        self.add_listener(surface.toplevel.set_title_event, self._on_set_title)
+        self.add_listener(surface.toplevel.set_app_id_event, self._on_set_app_id)
+        self.add_listener(handle.request_maximize_event, self._on_foreign_request_maximize)
+        self.add_listener(handle.request_minimize_event, self._on_foreign_request_minimize)
+        self.add_listener(handle.request_activate_event, self._on_foreign_request_activate)
+        self.add_listener(handle.request_fullscreen_event, self._on_foreign_request_fullscreen)
+        self.add_listener(handle.request_close_event, self._on_foreign_request_close)
+
+        self.qtile.manage(self)
+
+        # Send a frame done event to provide an opportunity to redraw even if we aren't
+        # going to map (i.e. because the window was opened on a hidden group)
+        surface.surface.send_frame_done(Timespec.get_monotonic_time())
 
     @expose_command()
     def kill(self) -> None:
+        self.hide()
```

Comment:
any reason why this is added? Removing this solves the kitty issue i'm having

Reply:
I think this can be removed. How has testing been? Any side effects with killing other clients?